### PR TITLE
Refine exception detail with i18n

### PR DESCRIPTION
**Pull request by JohnNiang**
#### What type of PR is this?

/kind feature
/kind api-change
/area core
/milestone 2.1.x

#### What this PR does / why we need it:

- Configuring message source location and name enables i18n message resolution.
- Simple global error handler.
- Refactor some exceptions with `ResponseStatusException` to control what HTTP status and problem detail need to be returned.

**TODO**

- [x] Add more UTs.
#### Which issue(s) this PR fixes:

Fixes https://github.com/halo-dev/halo/issues/3020

#### Special notes for your reviewer:

Steps to test:

1. Try to refine `src/main/resources/config/i18n/messages_zh.properties` and switch Browser language with Chinese.
2. Delibrately make a mistake as you wish and see the error tips in console.
3. Try to access one page which doesn't exist and see the rendered result.

#### Does this PR introduce a user-facing change?

```release-note
完成系统异常的国际化
```

### CODE REVIEW

File: src/main/resources/config/i18n/messages_zh.properties [hunk 1]
```diff
@@ -0,0 +1,4 @@
+problemDetail.title.org.springframework.web.server.ServerWebInputException=请求参数有误
+problemDetail.title.run.halo.app.infra.exception.AttachmentAlreadyExistsException=附件已存在
+
+problemDetail.run.halo.app.infra.exception.AttachmentAlreadyExistsException=文件 {0} 已存在，建议更名后重试。
```
Comment (ruibaby) [inline src/main/resources/config/i18n/messages_zh.properties:4]: 这个不和 message.properties 同步吗

**Review by ruibaby — CHANGES_REQUESTED** [review]

File: src/main/resources/config/i18n/messages_zh.properties (hunk 1 above)
Comment (JohnNiang) [reply src/main/resources/config/i18n/messages_zh.properties:4]: 关于中文翻译，后面可以慢慢补充。

**Review by ruibaby — COMMENTED** [review]
/lgtm

**Review by ruibaby — APPROVED** [review]

**Comment by guqing** [comment]
?? 就合并了

**Comment by guqing** [comment]
此 RP 将导致主题异常模板失效，始终只会渲染 Core 提供的默认异常模板

**Pull request merged**